tensorflow/stream_executor/cuda/cuda_blas.cc

```diff
@@ -2069,10 +2242,10 @@ bool CUDABlas::DoBlasGemmWithAlgorithmImpl(
   static_assert(blas::kDefaultGemmAlgo == CUBLAS_GEMM_DFALT, "");
   bool result = DoBlasInternalFailureOK(
       wrap::cublasGemmEx, stream, /* pointer_mode_host = */ true,
-      CUDABlasTranspose(transa), CUDABlasTranspose(transb), m, n, k, &alpha,
-      CUDAMemory(a), cuda_in_type, lda, CUDAMemory(b), cuda_in_type, ldb, &beta,
-      CUDAMemoryMutable(c), CUDADataType<OutT>::type, ldc,
-      CUDAComputationType(computation_type),
+      false /* = use_tensor_ops */, CUDABlasTranspose(transa),
```

Comment:
This is the part that the matmul autotuning works. Should we add that to part of the autotuning? 

DoBlasGemmWithAlgorithmImpl

This feature is currently disabled by default through this env-var "TF_MATMUL_AUTOTUNE_ENABLE", due to large noise. But our plan is to enable that. So it is a good idea to make sure it still works.

Reply:
I'd suggest that we add tensor_op support here, while include the setting in autotune in TensorFlow's op level by adding ways to propagate use_tensor_ops option to matmul op.

Reply:
Added cuda 9.0 algos, including tensor ops, to the algorithm list. Also, added checks to DoBlasGemmWithAlgorithmImpl to return false if tensor_op algorithm is requested when not appropriate. These changes enable the auto-tuning path to use tensor-ops.